lib/Service/DocumentService.php

```diff
@@ -193,8 +193,12 @@ public function addStep($documentId, $sessionId, $steps, $version): array {
 		}
 		// If there were any queries in the steps send the entire history
 		$getStepsSinceVersion = count($querySteps) > 0 ? 0 : $version;
+		// TODO: To discuss
+		// above causes increasing requests, maybe we need to filter out query steps in the above response
+		// Likely steps returned here will also get returned in the next fetch request, so maybe we just skip this
+		// as this was only a shortcut for the old handling to avoid another conflict/sync/push
```

Comment:
- Could be the old awareness messages that cause issues here where old awareness messages come in and update the state again
  - This PR: Maybe we can also filter awareness steps before returning them here
  - Follow up: Maybe store the last awareness message in oc_text_sessions/distributed memory cache (per session or generally per document)
  -